gremlin-go/driver/client.go

```diff
@@ -71,25 +73,51 @@ func NewClient(url string, configurations ...func(settings *ClientSettings)) (*C
 		transporterType: settings.TransporterType,
 		connection:      conn,
 	}
+	// TODO: PoolSize must be 1 on Session mode
 	return client, nil
 }
 
 // Close closes the client via connection.
 func (client *Client) Close() error {
-	return client.connection.close()
+	// If it is a Session, call closeSession
+	if client.Session != "" {
+		_, err := client.closeSession()
+		if err != nil {
+			return err
+		}
+	}
+	client.logHandler.logf(Info, "Closing Client with url %s", client.url)
+	err := client.connection.close()
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
In this last step we can omit the `if err != nil` check and just return `client.connection.close()`